storage/blob.go

```diff
@@ -513,6 +513,57 @@ func (b BlobStorageClient) GetBlob(container, name string) (io.ReadCloser, error
 	return resp.body, nil
 }
 
+// GetBlob returns a stream to read the blob. Caller must call Close() the
+// reader to close on the underlying connection.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179440.aspx
+func (b BlobStorageClient) GetBlobAndProperties(container, name string) (io.ReadCloser, *BlobProperties, error) {
```

Comment:
Alternatively it can be an unexported `getProperties` that is used both by this method and `GetBlobProperties`.